environmentManager/index.js

```diff
@@ -3,78 +3,109 @@ const { validEnvironment, validEnvironmentConfig } = require('../utils/parserCon
 
-    const environmentConfigs = validEnvironmentConfig(SPLIT_EVALUATOR_ENVIRONMENTS);
+      const environmentConfigs = validEnvironmentConfig(SPLIT_EVALUATOR_ENVIRONMENTS);
 
-    environmentConfigs.forEach(environment => {
+      environmentConfigs.forEach(environment => {
 
-      validEnvironment(environment);
-      const authToken = environment['AUTH_TOKEN'];
-      const apiKey = environment['API_KEY'];
-      settings.core.authorizationKey = apiKey;
+        validEnvironment(environment);
+        const authToken = environment['AUTH_TOKEN'];
+        const apiKey = environment['API_KEY'];
+        settings.core.authorizationKey = apiKey;
 
-      this._environments[authToken] = {
-        apiKey: apiKey,
-        factory: getSplitFactory(settings),
-        isClientReady: false,
-      };
+        this._environments[authToken] = {
+          apiKey: apiKey,
+          factory: getSplitFactory(settings),
+          isClientReady: false,
+        };
 
-      this._clientReadiness(this._environments[authToken].factory.client());
+        const client = this._environments[authToken].factory.client();
+        this._clientReadiness(client);
 
-    });
-    this.ready();
-  }
 
-  _clientReadiness(client){
-    this._readyPromises.push(client.ready());
-    client.on(client.Event.SDK_READY, () => client.isClientReady = true);
-  }
+      });
+      this.ready();
+    }
 
-  getFactory(authToken) {
-    return this._environments[authToken].factory;
-  }
+    _clientReadiness(client){
+      this._readyPromises.push(client.ready());
+      client.on(client.Event.SDK_READY, () => client.isClientReady = true);
+    }
 
-  getVersion() {
-    return this._environments[this.getAuthTokens()[0]].factory.settings.sdkVersion;
-  }
+    getFactory(authToken) {
+      return this._environments[authToken].factory;
+    }
 
-  getClient(authToken) {
-    return this._environments[authToken].factory.client();
-  }
+    getVersion() {
+      return this._environments[this.getAuthTokens()[0]].factory.settings.sdkVersion;
+    }
 
-  getManager(authToken) {
-    return this._environments[authToken].factory.manager();
-  }
+    getClient(authToken) {
+      return this._environments[authToken].factory.client();
+    }
 
-  validToken(authToken) {
-    return Object.keys(this._environments).indexOf(authToken) > -1;
-  }
+    getManager(authToken) {
+      return this._environments[authToken].factory.manager();
+    }
 
-  getAuthTokens() {
-    return Object.keys(this._environments);
-  }
+    validToken(authToken) {
+      return Object.keys(this._environments).indexOf(authToken) > -1;
+    }
 
-  ready() {
-    return Promise.allSettled(this._readyPromises).then(this._clientsReady = true);
-  }
+    getAuthTokens() {
+      return Object.keys(this._environments);
+    }
+
+    async ready() {
+      await Promise.all(this._readyPromises).then(this._clientsReady = true);
+    }
+
+    async destroy() {
+      this.getAuthTokens().forEach(async authToken => {
```

Comment:
This way the `destroy` result is resolved immediately because it is not waiting for individual destroy calls to resolve.

It should wait for individual destroy calls to resolve, by using either `await Promise.all(...)` or a for-look: check [here](https://stackoverflow.com/a/37576787/5824309)